### PR TITLE
[Form editor] Drag to reorder form fields

### LINKED ISSUE #226: [Form editor] Drag to reorder form fields
element `index` is used to represent field order:
https://github.com/google/ground-platform/wiki/Cloud-Firestore-Representation#document-structure

image

image

### PR DESCRIPTION
**Pull request by parulraheja98**
Closes #226 
image

### CODE REVIEW

File: web-ng/src/app/services/data-store/data-store.service.ts [hunk 1]
```diff
@@ -228,12 +228,13 @@ export class DataStoreService {
 
   private static formToJS(form: Form): {} {
     const { fields, ...formDoc } = form;
+    let index = 0;
```
Comment (gino-m) [inline web-ng/src/app/services/data-store/data-store.service.ts:231]: Could you please extract the call to `reduce` into its own named method to help make it clearer that this is where indices are reassigned here based on order in the form?

General question - do we ever sort the fields by index when displaying the fields in the form editor?

**Comment by parulraheja98** [comment]
@gino-m , here is the updated UI after drag icon change
image

File: web-ng/src/app/services/data-store/data-store.service.ts (hunk 1 above)
Comment (parulraheja98) [reply web-ng/src/app/services/data-store/data-store.service.ts]: Good question, right now the forms are not being rendered in the UI and only new forms are being created. I am going to make my next PR for rendering existing forms and ability to create new form using tabular switch between forms. I will consider sorting in that PR.

Comment (gino-m) [reply web-ng/src/app/services/data-store/data-store.service.ts]: Ok, will let this through for now. Thanks for creating a bug to track.

**Comment by gino-m** [comment]
@parulraheja98 could you kindly resolve conflict so we can merge? Sorry I didn't merge it yet, I was waiting for checks to pass and forgot to come back to it!

**Comment by parulraheja98** [comment]
Resolved conflicts, @gino-m

File: web-ng/src/app/components/layer-dialog/layer-dialog.component.css [hunk 2]
```diff
@@ -152,3 +151,11 @@
 .trash-icon {
   color: #707074;
 }
+
+.drag-icon {
+  display: block;
+  margin: 0px auto;
+  cursor: move;
+  padding: 5px;
+  color: #dadde1;
```
Comment (gino-m) [inline web-ng/src/app/components/layer-dialog/layer-dialog.component.css:160]: Could we turn this into scss and use `mat-color($foreground, lighter)` instad?

I was trying to follow https://material.angular.io/guide/theming-your-components but was unsuccessful.. Any idea how to set that up?

**Comment by gino-m** [comment]
💯, great work @parulraheja98 !

**Pull request merged**